File: JavaCore_TestingSystem_Assignment_6/src/com/vti/entity/Department.java
package com.vti.entity;

import utils.ScannerUtils;

public class Department {
	public static int COUNT = 0;
	private int id;
	private String name;

	// Constructor có tham số
	public Department(String name) {
		super();
		COUNT++;
		this.id = COUNT;
		this.name = name;
	}

	// Constructor không tham số - tự động nhập
	public Department() {
		super();
		COUNT++;
		this.id = COUNT;
		System.out.print("Nhập tên phòng: ");
		this.name = ScannerUtils.inputString();
	}

	@Override
	public String toString() {
		return "Department [id=" + id + ", name=" + name + "]";
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}
}

File: Fontend_HTML_CSS/ReactJS/1. TypeScript/LAB2_Testing_System_OOP/Department.js
class Department {
    id;
    name;
    constructor(id, name) {
        this.id = id;
        this.name = name;
    }
    getId() {
        return this.id;
    }
    getName() {
        return this.name;
    }
    setId(id) {
        this.id = id;
    }
    setName(name) {
        this.name = name;
    }
}
// Khai báo 1 function để in thông tin các phòng ban.
function printDepartment(department) {
    console.log(`ID: ${department.getId()}, Name: ${department.getName()}`);
}
export { Department, printDepartment };
